quick-dsp/src/bell202/mod.rs

```diff
@@ -93,11 +94,65 @@ where
         .apply_one_to_one(Decimator::<f32, Out>::default())
 }
 
+pub struct Ax25Debug<'a>(pub &'a [u8]);
+
+impl<'a> Ax25Debug<'a> {
+    /// Returns true if this looks like a AX25 packet
+    pub fn is_ax25(&self) -> bool {
+        let mut addr_len = 0u32;
+        for x in self.0.iter() {
+            let x = *x;
+            addr_len += 1;
+            if x & 1 == 1 {
+                break;
+            }
+        }
+
+        addr_len > 10
+    }
+}
+
+impl<'a> Debug for Ax25Debug<'a> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        use core::str::from_utf8;
+        let mut addrfield = vec![];
+
+        for x in self.0.iter() {
```

Comment:
Perhaps, `self.0.iter().take_while(|&x| { addrfield.push(x >> 1); x & 1 != 1});` though perhaps that's a bit too clever.